service/internal/checksum/middleware_compute_input_checksum_test.go

```diff
@@ -968,6 +968,246 @@ func TestComputeInputPayloadChecksum(t *testing.T) {
+					return r
+				}(),
+			},
+			expectHeader: http.Header{
+				"X-Amz-Checksum-Crc32": []string{"DUoRhQ=="},
+			},
+			expectPayload: []byte("hello world"),
+			expectChecksumMetadata: map[string]string{
+				"CRC32": "DUoRhQ==",
+			},
+			expectErr: "mock error in first attempt",
+		},
+		"http request with stream updated": {
+			initContext: func(ctx context.Context) context.Context {
+				return internalcontext.SetChecksumInputAlgorithm(ctx, string(AlgorithmCRC32))
+			},
+			buildInput: middleware.BuildInput{
+				Request: func() *smithyhttp.Request {
+					r := smithyhttp.NewStackRequest().(*smithyhttp.Request)
+					r.URL, _ = url.Parse("http://example.aws")
+					r.ContentLength = 11
+					r = requestMust(r.SetStream(bytes.NewReader([]byte("hello world"))))
+					return r
+				}(),
+			},
+			updatedStreamBody: "world hello",
+			expectHeader: http.Header{
+				"X-Amz-Checksum-Crc32": []string{"DUoRhQ=="},
+			},
+			expectPayload: []byte("world hello"),
+			expectChecksumMetadata: map[string]string{
+				"CRC32": "DUoRhQ==",
+			},
+			expectErr: "mock error in first attempt",
+		},
+		"https request with stream unchanged": {
+			initContext: func(ctx context.Context) context.Context {
+				return internalcontext.SetChecksumInputAlgorithm(ctx, string(AlgorithmCRC32))
+			},
+			buildInput: middleware.BuildInput{
+				Request: func() *smithyhttp.Request {
+					r := smithyhttp.NewStackRequest().(*smithyhttp.Request)
+					r.URL, _ = url.Parse("https://example.aws")
+					r.ContentLength = 11
+					r = requestMust(r.SetStream(bytes.NewReader([]byte("hello world"))))
+					return r
+				}(),
+			},
+			expectHeader: http.Header{
+				"Content-Encoding": []string{"aws-chunked"},
+				"X-Amz-Trailer":    []string{"x-amz-checksum-crc32"},
+			},
+			expectPayload:         []byte("b\r\nhello world\r\n0\r\nx-amz-checksum-crc32:DUoRhQ==\r\n\r\n"),
+			expectDeferToFinalize: true,
+			expectChecksumMetadata: map[string]string{
+				"CRC32": "DUoRhQ==",
+			},
+			expectErr: "mock error in first attempt",
+		},
+		"https request with stream updated": {
+			initContext: func(ctx context.Context) context.Context {
+				return internalcontext.SetChecksumInputAlgorithm(ctx, string(AlgorithmCRC32))
+			},
+			buildInput: middleware.BuildInput{
+				Request: func() *smithyhttp.Request {
+					r := smithyhttp.NewStackRequest().(*smithyhttp.Request)
+					r.URL, _ = url.Parse("https://example.aws")
+					r.ContentLength = 11
+					r = requestMust(r.SetStream(bytes.NewReader([]byte("hello world"))))
+					return r
+				}(),
+			},
+			updatedStreamBody: "world hello",
+			expectHeader: http.Header{
+				"Content-Encoding": []string{"aws-chunked"},
+				"X-Amz-Trailer":    []string{"x-amz-checksum-crc32"},
+			},
+			expectPayload:         []byte("b\r\nworld hello\r\n0\r\nx-amz-checksum-crc32:DUoRhQ==\r\n\r\n"),
+			expectDeferToFinalize: true,
+			expectChecksumMetadata: map[string]string{
+				"CRC32": "DUoRhQ==",
```

Comment:
Why would the updated stream body have the same checksum as the original request?

Reply:
The SEP didn't say which value should we set in metadata, but imo we should always set the original value here, if the payload is updated between retry, error will be returned and picked up so metadata value doesn't matter in this case